dashboard/src/shared/Auth.ts

```diff
@@ -60,21 +65,27 @@ export class Auth {
   // Throws an error if the token is invalid
   public static async validateToken(cluster: string, token: string) {
     try {
-      await Axios.get(url.api.k8s.base(cluster) + "/", {
-        headers: { Authorization: `Bearer ${token}` },
+      await this.resourcesClient(token).CheckNamespaceExists({
```

Comment:
+1 to this approach; however, I wonder whether we should use an alternative function here. I mean, at first glance, calling `CheckNamespaceExists` for the `validateToken` function seems a bit misleading.

Perhaps we could use the `can-i` operation there or, even better, directly use the k8s object aimed at validating tokens, the `TokenReview` one ([more info](https://kubernetes.io/docs/reference/access-authn-authz/authentication/)):


```json
{
  "apiVersion": "authentication.k8s.io/v1",
  "kind": "TokenReview",
  "spec": {
    "token": "014fbff9a07c...", 
  }
}
```

Reply:
Yeah, I'd initially gone along this direction, adding a new endpoint to the resources plugin (didn't think of using the TokenReview - that would have been a good match), but the reason I stopped was that this functionality (for `validateToken`) is relevant only for the "demo-only" functionality of using service account tokens to authenticate with Kubeapps.

The existing functionality just tries to get the root URL of the k8s API server and I realised that the endpoint is actually arbitrary - the check just needs to see a 200 or 403 - either is fine.

+1 to switch it over to a `can-i` endpoint, once we have one available in the resources plugin.